metals/src/main/scala/scala/meta/internal/metals/BuildServerConnection.scala

```diff
@@ -180,6 +180,20 @@ class BuildServerConnection private (
       .map(address => URI.create(address.getUri))
   }
 
+  def jvmRunEnvironment(
+      params: JvmRunEnvironmentParams
+  ): Future[JvmRunEnvironmentResult] = {
+    if (initialConnection.capabilities.getJvmRunEnvironmentProvider()) {
```

Comment:
checking `initialConnection` bothers me a bit since we can reconnect to the bsp, but build server stays the same thus capabilities shouldn't change.

Reply:
Yeah, that can't really change. But reworked it to use `connection` which will be a completed future in most cases.